### ISSUE TITLE
Issue locating/using elements within an iFrame using SelIDE

### ISSUE THREAD

**Issue by 98gmarquee** [SeleniumHQ/selenium-ide#423]
## 🐛 Bug Report

I'm unable to focus or interact with buttons/links/elements in an iframe via ff quantum or chrome using the current IDE.

## To Reproduce
Access a page that has buttons/links inside an iframe without an id, only src attribute and attempt to click these buttons. It's unable to find them. 

Also try using select frame using xpath to locate, such as
`select frame| xpath=//iframe[@src='/foo/foo.jsp'| `

Also tried recording and it recorded a command such as:
`select frame|Index=0 | `
Then when I try running it, it gets stuck on that line and throws an error.

It cannot find the frame to then focus on to run commands in that frame.

## Expected behavior

It should be able to find the elements within the frame.

## Project file reproducing this issue (highly encouraged)
I cannot produce the side file as it's accessing an internal web site. I've attached an image of the source for the iFrame including the button I'm attempting to locate/click highlighted. Hopefully that is sufficient.
![selide-iframeissue](https://user-images.githubusercontent.com/33339926/46160417-4c663280-c248-11e8-888a-67711227bbcb.png)

That is of this button:
![selide-iframeissue-screen](https://user-images.githubusercontent.com/33339926/46161439-2b531100-c24b-11e8-9fc7-15c52418920d.png)

## Environment
OS: w10
Selenium IDE Version:  3.3.1
Selenium SIDE Runner Version:  n/a
Node version: n/a
Browser:
Browser Version: firefox 62.0.2 64bit, chrome  69.0.3497.100 64bit

**Comment by luquitia** [comment]
I'm also having this issue. I'm not sure how to structure the argument for **select frame** command; the reference tab is vague on this point (and the mentioned index options don't work for me).

I have an iframe with attributes like so: <iframe scrolling="no" allowtransparency="true" id="profileBuilder" title="Create Your Profile" style="border: 0px none; height: 297px;" src="/LongURL-thatIremovedHere" tabindex="0">

I tried:
select frame | id=profileBuilder
select frame | index=0
select frame | index=1
select frame | xpath=//iframe[@id='profileBuilder']
and some other variations
-- all throw errors like:  Failed: Invalid argument or Failed: Timeout

_Note: recording doesn't work for me as the playback doesn't recognize clicks within this iframe (since I guess it is not "selected")._

**Comment by jngale** [comment]
This seems to be something that was introduced in the last few months. I have items from May where this was not happening. Is there any eta on this?

**Comment by tourdedave** [comment]
Is the target frame the only one or is there more than one (e.g., one inside of another)?

**Comment by 98gmarquee** [comment]
Using index=0 seems to work now for me. However, even with chrome, I'm getting an error selecting a file to import on the site, so I can't tell if anything works after that or how to select the main page as the "next" button to navigate to the next page is outside the frame. On FF, it gave me a message saying that opening a file only works in chrome currently, so I tried chrome and it can't find the input field for the file even though I see it get highlighted when viewing the source in dev tools.

It's an issue outside this one. I'll look into any other file handling issues to see if I can figure out any tips to deal with this.

**Comment by jngale** [comment]
@tourdedave For us the target is the only one.

**Comment by tourdedave** [comment]
@98gmarquee Glad to hear an index locators for you. If you're unable to resolve your file open issue please open a new issue.

@jngale Is the issue still happening for you?

**Comment by jngale** [comment]
Yes, it still seems to be happening

**Comment by lievi** [comment]
There an update about this issue.

I had the same error in my script. The relative=top not working and i cannot got to the main frame after enter in someone.

**Comment by gherkz** [comment]
If anyone else is struggling with selecting the top frame after selecting a subframe, I found the answer by trawling through the source code:

https://github.com/SeleniumHQ/selenium-ide/blob/deba17d4d2444fbef28433d57438fe42779990b0/packages/selenium-ide/src/neo/IO/SideeX/ext-command.js#L429

TL;DR:
Unless I am misunderstanding this match, `relative=top` doesn't appear to be recognized as a valid target, even though the docs say it is. `relative=parent` seemed to work for me though

**Comment by tourdedave** [comment]
To sum this up -- it looks like the recorder only captures `relative=parent`, but we document `relative=top` as a viable option in the `select frame`, but it's not a supported locator as part of playback.

This should work in playback. It's a bug. Marking as confirmed. I'll get a fix in for it soon.

**Comment by tourdedave** [comment]
This will go out in the next release.
